src/metaTypes.h

```diff
@@ -35,18 +36,18 @@ namespace METAIO_NAMESPACE {
 typedef char                MET_ASCII_CHAR_TYPE;
 typedef char                MET_CHAR_TYPE;
 typedef unsigned char       MET_UCHAR_TYPE;
-typedef short               MET_SHORT_TYPE;
-typedef unsigned short      MET_USHORT_TYPE;
-typedef int                 MET_INT_TYPE;
-typedef unsigned int        MET_UINT_TYPE;
-typedef long                MET_LONG_TYPE;
-typedef unsigned long       MET_ULONG_TYPE;
+typedef int16_t             MET_SHORT_TYPE;
+typedef uint16_t            MET_USHORT_TYPE;
+typedef int32_t             MET_INT_TYPE;
+typedef uint32_t            MET_UINT_TYPE;
+typedef int32_t             MET_LONG_TYPE;
+typedef uint32_t            MET_ULONG_TYPE;
```

Comment:
Hi,

MET_ULONG_TYPE could be called MET_FOO.   It is unfortunate that the name
doesn't match the construct commonly used now, but it really is ok.

I'd be up for changing the name - but if we do change it, we should
probably change it drastically, so that anyone who was using MET_ULONG_TYPE
will get a compilation error.   For example, we could eliminate
MET_ULONG_TYPE and instead have MET_INT64_TYPE and MET_UINT46_TYPE.

s


On Wed, Sep 14, 2016 at 11:00 AM, Bradley Lowekamp <notifications@github.com
> wrote:

> In src/metaTypes.h
> <https://github.com/Kitware/MetaIO/pull/22#discussion_r78766417>:
>
> > @@ -35,18 +36,18 @@ namespace METAIO_NAMESPACE {
> >  typedef char                MET_ASCII_CHAR_TYPE;
> >  typedef char                MET_CHAR_TYPE;
> >  typedef unsigned char       MET_UCHAR_TYPE;
> > -typedef short               MET_SHORT_TYPE;
> > -typedef unsigned short      MET_USHORT_TYPE;
> > -typedef int                 MET_INT_TYPE;
> > -typedef unsigned int        MET_UINT_TYPE;
> > -typedef long                MET_LONG_TYPE;
> > -typedef unsigned long       MET_ULONG_TYPE;
> > +typedef int16_t             MET_SHORT_TYPE;
> > +typedef uint16_t            MET_USHORT_TYPE;
> > +typedef int32_t             MET_INT_TYPE;
> > +typedef uint32_t            MET_UINT_TYPE;
> > +typedef int32_t             MET_LONG_TYPE;
> > +typedef uint32_t            MET_ULONG_TYPE;
>
> Well I have not clue how MetaIO works when it thinks MET_ULONG_TYPE is 4
> bytes, but it's actually 8!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/MetaIO/pull/22/files/106a72fe8beba785e9ba98c35bb3f2cc32f94618#r78766417>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAREr8J21bOKc2cLuZsW7l5Kbv3Ok-HDks5qqAwDgaJpZM4J8EAc>
> .
>



-- 
====================================================
Stephen R. Aylward, Ph.D.
Senior Director of Operations, North Carolina, Kitware, Inc.
http://www.kitware.com and http://www.aylward.org
(919) 969-6990 x300
====================================================